File: output/ChatGPT/codenet/Python/C++/s575946485.cpp
#include <iostream>
#include <vector>
#include <map>
#include <algorithm>
using namespace std;

int main() {
    vector<int> a;
    for (int i = 0; i < 5; i++) {
        int input;
        cin >> input;
        a.push_back(input);
    }

    map<int, int> dic;

    for (int i = 0; i < 5; i++) {
        if (to_string(a[i]).back() == '0') {
            dic[i] = 0;
        } else {
            dic[i] = stoi(to_string(a[i]).back()) - 10;
        }
    }

    int min_num = min_element(dic.begin(), dic.end(), [](const pair<int, int>& p1, const pair<int, int>& p2) {
        return p1.second < p2.second;
    })->first;

    int a_sum = accumulate(a.begin(), a.end(), 0);
    dic.erase(min_num);
    cout << abs(accumulate(dic.begin(), dic.end(), 0, [](int sum, const pair<int, int>& p) {
        return sum + p.second;
    })) + a_sum;
    

    return 0;
}

File: output/CodeGen/codenet/Python/Java/s575946485.java
a = new int[]{1, 2, 3, 4, 5};

dic = new HashMap<Integer, Integer>();

for (int i = 0; i < a.length; i++) {
    if (a[i] % 10 == 0) {
        dic.put(i, 0);
    } else {
        dic.put(i, a[i] % 10);
    }
}

int min_num = 0;
int a_sum = 0;
for (int i = 0; i < dic.size(); i++) {
    if (dic.get(i) == min(dic.values())) {
        min_num = i;
    }
}

a_sum = a[0] + a[1] + a[2] + a[3] + a[4];
dic.remove(min_num);

int sum = 0;
for (int i = 0; i < dic.size(); i++) {
    sum += dic.get(i);
}

System.out.println(Math.abs(sum + a_sum));